src/vote_graph.rs

```diff
@@ -304,7 +304,7 @@ impl<H, N, V> VoteGraph<H, N, V> where
 					}
 				})
 				.filter(|&(_, ref node)| condition(&node.cumulative_vote))
-				.next();
+				.max_by(|&(_, ref node1), &(_, ref node2)| node1.number.cmp(&node2.number));
```

Comment:
As mentioned in a Riot discussion, the two statements are actually equivalent. Note this line in the docs above:

https://github.com/paritytech/finality-grandpa/blob/12482fcbf36873e8c787e0012e9d25f5b36ec4af/src/vote_graph.rs#L261-L263